decorator.rb

```diff
@@ -0,0 +1,5 @@
+class Nameable
+  def correct_name
+    raise NotImplementedError
+  end
```

Comment:
- [ ] [OPTIONAL] The error message can be made more specific raise `NotImplementedError, "#{self.class} has not implemented method '#{__method__}'"`